src/main/java/org/tinyradius/io/server/SecretProvider.java

```diff
@@ -12,5 +14,17 @@
      * @return shared secret or null
      */
     String getSharedSecret(InetSocketAddress address);
+    
+    /**
+     * An alternative method of returning shared secret but this time with the current
+     * radius request so that alternative secrets can be returned based on some sort of 
+     * context. 
+     * 
+     * By default this method calls the original getSharedSecret.
+     * @param address
+     * @param request
+     * @return
```

Comment:
Could we fill in the javadoc fields+description, even if it's just copying it from the overloaded method?

Reply:
Fixed